server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/DeadCompactionDetector.java

```diff
@@ -145,9 +146,14 @@ private void detectDeadCompactions() {
     Map<ExternalCompactionId,KeyExtent> tabletCompactions = new HashMap<>();
 
     // find what external compactions tablets think are running
-    try (TabletsMetadata tabletsMetadata = context.getAmple().readTablets().forLevel(DataLevel.USER)
-        .filter(new HasExternalCompactionsFilter()).fetch(ColumnType.ECOMP, ColumnType.PREV_ROW)
-        .build()) {
+    try (Stream<TabletMetadata> tabletsMetadata = Stream
+        // Listing the data levels vs using DataLevel.values() prevents unexpected
+        // behavior if a new DataLevel is added
+        .of(DataLevel.ROOT, DataLevel.METADATA, DataLevel.USER)
+        .map(dataLevel -> context.getAmple().readTablets().forLevel(dataLevel)
```

Comment:
Uncertain if this will close the Closeable returned by context.getAmple().readTablets()

Reply:
This is addressed in #4357